plugin/hotswap-agent-spring-plugin/src/main/java/org/hotswap/agent/plugin/spring/core/BeanFactoryProcessor.java

```diff
@@ -0,0 +1,131 @@
+package org.hotswap.agent.plugin.spring.core;
+
+import org.hotswap.agent.logging.AgentLogger;
+import org.hotswap.agent.plugin.spring.transformers.api.IPlaceholderConfigurerSupport;
+import org.hotswap.agent.plugin.spring.utils.AnnotatedBeanDefinitionUtils;
+import org.hotswap.agent.util.ReflectionHelper;
+import org.hotswap.agent.util.spring.util.ReflectionUtils;
+import org.springframework.beans.factory.annotation.AnnotatedBeanDefinition;
+import org.springframework.beans.factory.config.BeanDefinition;
+import org.springframework.beans.factory.config.PlaceholderConfigurerSupport;
+import org.springframework.beans.factory.support.*;
+import org.springframework.context.annotation.ScannedGenericBeanDefinition;
+import org.springframework.context.support.GenericApplicationContext;
+import org.springframework.util.StringValueResolver;
+
+import java.lang.reflect.Constructor;
+import java.lang.reflect.Field;
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.util.List;
+import java.util.function.Predicate;
+
+public class BeanFactoryProcessor {
+
+    private static AgentLogger LOGGER = AgentLogger.getLogger(BeanFactoryProcessor.class);
+
+    public static void removeBeanDefinition(DefaultListableBeanFactory beanFactory, String beanName) {
+        resetEmbeddedValueResolvers(beanFactory, beanName);
+        beanFactory.removeBeanDefinition(beanName);
+    }
+
+    public static void destroySingleton(DefaultListableBeanFactory beanFactory, String beanName) {
+        // remove embeddedValueResolvers cache in PlaceholderConfigurerSupport
+        resetEmbeddedValueResolvers(beanFactory, beanName);
+        beanFactory.destroySingleton(beanName);
+    }
+
+    public static boolean isFactoryBean(DefaultListableBeanFactory beanFactory, String beanName, RootBeanDefinition beanDefinition) {
```

Comment:
Not sure why isFactoryBean(String) method not work here.

```java
beanFactory.isFactoryBean(beanName);
```

Reply:
It is not public method.

Reply:
isFactoryBean is now useless, can we simply remove it?

Reply:
Temporary remove the code